src/redux/notifications/reducer.ts

```diff
@@ -1,13 +1,65 @@
 import { INotificationsState } from './state';
 import { IAction } from '../types';
+import { SET_UNSEEN_NOTIFICATIONS, SET_NOTIFICATIONS, MARK_SEEN } from './actions';
 
-const initialState: INotificationsState = {};
+const initialState: INotificationsState = {
+    unseenNotifications: 0,
+    notifications: {}
+};
 
 export default (
     state: INotificationsState = initialState,
     action: IAction
 ): INotificationsState => {
     switch (action.type) {
+        case SET_UNSEEN_NOTIFICATIONS:
+            return {
+                ...state,
+                unseenNotifications: action.data.unseenNotifications
+            };
+
+        case SET_NOTIFICATIONS: {
+            let notifications = {};
+
+            action.data.notifications.map((notif: any) => {
+                const notifData = {
+                    walletId: notif.walletId,
+                    title: notif.title,
+                    body: notif.body,
+                    seen: notif.seen,
+                    data: notif.data
+                };
+
+                notifications = {
+                    ...notifications,
+                    [notif.data.blockchain]: {
+                        ...notifications[notif.data.blockchain],
+                        [notif._id]: notifData
```

Comment:
is ok to use notif._id ? - _id is not something only used by mongo, and we use notifId?